[CI] HnswUtilsTests testNeighborhoodGraphSearchFindsSelf failing

**Build Scans:**
- [elasticsearch-intake #44188 / part1](https://gradle-enterprise.elastic.co/s/j4kempihs727m)
- [elasticsearch-pull-request #165430 / part-1](https://gradle-enterprise.elastic.co/s/lmmthrqrqiqda)
- [elasticsearch-pull-request #165430 / part-1](https://gradle-enterprise.elastic.co/s/5pjvywlkvzcgc)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.index.codec.vectors.HnswUtilsTests.testNeighborhoodGraphSearchFindsSelf" -Dtests.seed=3994AE8856F14D7B -Dtests.locale=shi-Latn-MA -Dtests.timezone=Europe/Ljubljana -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.codec.vectors.HnswUtilsTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testNeighborhoodGraphSearchFindsSelf'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: graph search found self in only 34/40 queries
```

**Issue Reasons:**
- [main] 3 failures in test testNeighborhoodGraphSearchFindsSelf (0.8% fail rate in 385 executions)
- [main] 2 failures in step part-1 (0.8% fail rate in 250 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -622,9 +622,6 @@ tests:
 - class: org.elasticsearch.upgrades.TokenBackwardsCompatibilityIT
   method: testTokensStayInvalidatedInUpgradedCluster
   issue: https://github.com/elastic/elasticsearch/issues/154627
-- class: org.elasticsearch.index.codec.vectors.HnswUtilsTests
-  method: testNeighborhoodGraphSearchFindsSelf
-  issue: https://github.com/elastic/elasticsearch/issues/154633
 - class: org.elasticsearch.xpack.stateless.StatelessHollowIndexShardsIT
   method: testStressWithRelocationOrObjectStoreFailures
   issue: https://github.com/elastic/elasticsearch/issues/154655

--- a/server/src/test/java/org/elasticsearch/index/codec/vectors/HnswUtilsTests.java
+++ b/server/src/test/java/org/elasticsearch/index/codec/vectors/HnswUtilsTests.java
@@ -26,6 +26,8 @@ import java.util.ArrayDeque;
 import java.util.ArrayList;
 import java.util.List;
 
+import static org.apache.lucene.util.hnsw.HnswGraphBuilder.DEFAULT_MAX_CONN;
+
 public class HnswUtilsTests extends ESTestCase {
 
     public void testGraphSerialization() throws IOException {
@@ -154,7 +156,7 @@ public class HnswUtilsTests extends ESTestCase {
             }
         }
         var sim = VectorSimilarityFunction.EUCLIDEAN;
-        int m = random().nextInt(8, 32);
+        int m = random().nextInt(DEFAULT_MAX_CONN, 32);
         int candidates = Math.min(numVectors - 1, 64);
         var neighborhoods = NeighborHood.computeNeighborhoods(CentroidOps.FLOAT, vectors, candidates);
         HnswUtils.MultiLevelAdjacency built = HnswUtils.buildMultiLevelFromNeighborhoods(neighborhoods, floatScorer(vectors, sim), m, 42L);

